cylc/flow/etc/job.sh

```diff
@@ -182,29 +241,42 @@ cylc__job__run_inst_func() {
 #     messages to send back to the suite server program,
 #     see "cylc help message" for format of messages.
 # Returns:
-#   exit 1
+#   exit ${CYLC_TASK_USER_SCRIPT_EXITCODE}
 cylc__job_finish_err() {
+    CYLC_TASK_USER_SCRIPT_EXITCODE="${CYLC_TASK_USER_SCRIPT_EXITCODE:-$?}"
     typeset signal="$1"
     typeset run_err_script="$2"
     shift 2
-    typeset signal_name=
     # (Ignore shellcheck "globbing and word splitting" warning here).
     # shellcheck disable=SC2086
     trap '' ${CYLC_VACATION_SIGNALS:-} ${CYLC_FAIL_SIGNALS}
     if [[ -n "${CYLC_TASK_MESSAGE_STARTED_PID:-}" ]]; then
         wait "${CYLC_TASK_MESSAGE_STARTED_PID}" 2>'/dev/null' || true
     fi
-    cylc message -- "${CYLC_SUITE_NAME}" "${CYLC_TASK_JOB}" "$@" || true
+    # Propagate real signals to entire process group, if we are a group leader,
+    # otherwise just to the backgrounded user script.
+    if [[ -n "${CYLC_TASK_USER_SCRIPT_PID:-}" ]] &&
+       [[ ! "${signal}" =~ '^(ERR|EXIT)$' ]]; then
```

Comment:
The quotes need to be removed from the right-hand side:
https://mywiki.wooledge.org/BashPitfalls#if_.5B.5B_.24foo_.3D.2BAH4_.27some_RE.27_.5D.5D

Reply:
@dpmatthews, even though I fixed the quoting, it looks like regex matching is causing bash 3.2 action to fail consistently. I've downloaded the action logs to see what's going on and there is syntax error thrown on this line. Dunno why, because bash 3.2 should support this syntax (it works if I try locally). Should I revert this to use glob matching as before? Or alternatively:

```bash
    if [[ -n "${CYLC_TASK_USER_SCRIPT_PID:-}" ]] &&
       [[ "${signal}" != "ERR" ]] && [[ "${signal}" != "EXIT" ]]; then
```

Reply:
Ha, that explains it - we were cross-posting :grin:

Reply:
(Jeez, you must be up late!)

Reply:
@TomekTrzeciak I can't explain this either but your alternative suggestion above looks fine to me.